### DISCUSSION TITLE
service account

### DISCUSSION THREAD

**Discussion by AlesyaSeliazniova30032012** [community/community#131131] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

can I create service account without email? Need to access git, need app account, not attached to any human

**Comment by chiarorosa** [comment]
@AlesyaSeliazniova30032012 Yes, you can create a service account for accessing GitHub without associating it with a specific email. Here’s how you can do it:

### 1. Create a Machine User
A machine user is a GitHub account that isn’t tied to a specific human. It’s commonly used for automation purposes, such as CI/CD pipelines.

1. **Sign Up for a New GitHub Account:**
   - Go to [GitHub's Sign Up Page](https://github.com/join).
   - Create a new account using a new email address (this can be a generic email or a shared email within your team).

2. **Set Up SSH Keys or Personal Access Token:**
   - You can use SSH keys or a Personal Access Token (PAT) to authenticate the machine user. This avoids using the account password.
   - [Generate a new SSH key](https://docs.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) and add it to your GitHub account.
   - Alternatively, [create a PAT](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) with the necessary scopes for your use case.

### 2. Use the Machine User in Your App
- **SSH Key Authentication:**
  ```sh
  ssh-keygen -t rsa -b 4096 -C "your_email@example.com"  # Generate SSH key
  eval "$(ssh-agent -s)"
  ssh-add ~/.ssh/id_rsa
  ```
  - Add the public key to the machine user’s GitHub account under `Settings` > `SSH and GPG keys`.

- **Personal Access Token Authentication:**
  - Create a PAT from `Settings` > `Developer settings` > `Personal access tokens`.

### 3. Automate with GitHub Actions or Other CI/CD Tools
Use the SSH key or PAT in your CI/CD pipeline. For example, in a GitHub Actions workflow, you can set the PAT as a secret and use it to authenticate:

```yaml
name: CI

on: [push]

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout repository
        uses: actions/checkout@v2
        with:
          token: ${{ secrets.GITHUB_TOKEN }}  # Use the token stored as a secret
```

### Notes:
- **Email Requirement:** Although you need an email to create the GitHub account, it doesn't have to be a personal one. You can use a generic or shared team email.
- **Repository Access:** Make sure the machine user has the necessary permissions to access the repositories. You can add it as a collaborator or give it access through a team in your organization.

By following these steps, you can create a service account for GitHub that isn’t tied to any specific individual, allowing for automated processes and better security management.

**Comment by AlesyaSeliazniova30032012** [comment]
@chiarorosa is it possible to create a service account without linking it to a specific email, so that, for example, any team member can access it?
